src/DynamoCore/Logging/DynamoAnalyticsClient.cs

```diff
@@ -258,6 +259,28 @@ public void TrackScreenView(string viewName)
                 }
             });
         }
+        /// <summary>
+        /// This API is used to track user/machine's activity status.
+        /// </summary>
+        /// <param name="activityType">Value must be either machine or user. If no value is provided the API will default to user activity type.</param>
+        public void TrackActivityStatus(string activityType)
+        {
+            if (Analytics.DisableAnalytics) return;
+
+            Task.Run(() =>
+            {
+                serviceInitialized.Wait();
+
+                lock (trackEventLockObj)
```

Comment:
hmm

@pinzart90 @BogdanZavu - What is this lock protecting?
`ReportingAnalytics`? Is that really going to be changing?

Because this task gets run when a user clicks the left mouse button it seems we're going to kick off a bunch of tasks that all lock and wait for each other ... is the lock really needed in this case?

Reply:
In debug build, it takes less than 1000 ticks.

Reply:
ReportingAnalytics? Is that really going to be changing? - mostlikely not but it is safer imo to have it like this. It does not buy us anything if it is outside, it will only exposes us to risk now and in the future if the implementation changes.

The lock is protecting from concurrent access into ADP api. In theory that is noted as being thread safe but there were reports with relatively simple use cases where surprisingly it is not. So to be on the safe side we have this lock.